core/blockchain.go

```diff
@@ -2034,6 +2021,20 @@ func (bc *BlockChain) insertChain(chain types.Blocks, setHead bool) (int, error)
 		if parent == nil {
 			parent = bc.GetHeader(block.ParentHash(), block.NumberU64()-1)
 		}
+
+		// check blob data available first
+		// TODO(GalaIO): move IsDataAvailable combine into verifyHeaders?
+		if bc.chainConfig.IsCancun(block.Number(), block.Time()) {
+			if posa, ok := bc.engine.(consensus.PoSA); ok {
+				blobs, exist := bc.receivedBlobsCache.Load(block.Hash())
+				if !exist {
```

Comment:
Are we going to save empty blobs as well?

Reply:
Yes, empty blobs means not blobs, same as empty transactions or receipts.